ntrans_gui.py

```diff
@@ -0,0 +1,268 @@
+            )
+            for n, var in self.checkbox_vars.items()
+        }
+
+        # What language to translate into
+        target_language_label = ttk.Label(mainframe, text="Target language")
+
+        self.target_language_var = tkinter.StringVar()
+
+        target_language = ttk.Combobox(
+            mainframe, state="readonly", textvariable=self.target_language_var
+        )
+
+        target_language["values"] = (
+            "German",
+            "Spanish",
+            "French",
+            "Swedish",
+            "Italian",
+        )
+
+        # How many of each N-Gram to translate
+        data_size_label = ttk.Label(mainframe, text="Amount of each N-gram to include")
+
+        self.data_size_var = tkinter.IntVar()
+
+        ngram_data_size = ttk.Combobox(
+            mainframe, state="readonly", textvariable=self.data_size_var
+        )
+
+        ngram_data_size["values"] = (
+            100,
+            300,
+            500,
+            800,
+            1000,
+            1500,
+            3000,
+            5000,
+            10000,
+        )
+
+        ngram_data_size.current(6)
+
+        # Generate Button
+        generate_button_style = ttk.Style()
+        generate_button_style.configure(
+            "W.TButton",
+            font=("TkDefaultFont", 16),
+            foreground="yellow",
+            background="black",
+            padding=30,
+        )
+        generate_dictionary = ttk.Button(
+            mainframe,
+            command=self.generate_ntrans_dictionary,
+            text="Generate N-Trans Dictionary",
+            style="W.TButton",
+        )
+
+        # About / Help
+        about_help_buttonframe = ttk.Frame(mainframe)
+
+        about_button = ttk.Button(
+            about_help_buttonframe,
+            command=self.open_about_window,
+            text="About",
+            style="",
+        )  # TODO: Make command into lambda instantiating about-class
+        help_button = ttk.Button(
+            about_help_buttonframe, command=self.open_help_page, text="Help"
+        )
+
+        # Black turn off formatting
+        # fmt: off
+
+        # ==================
+        # Widget placement
+        # ==================
+        header.grid(column=0, row=0, columnspan=2, padx=(0, 0), pady=(30, 30))
+
+        ngram_check_label.grid(sticky="N", column=0, row=1, padx=(20, 0), pady=(10, 10))
+
+        select_all_checkbutton.grid(sticky="W", column=0, row=2, padx=(40, 0), pady=(0, 10))
+        n_to_ngram_checkbox[2].grid(sticky="W", column=0, row=3, padx=(40, 0), pady=(0, 0))
+        n_to_ngram_checkbox[3].grid(sticky="W", column=0, row=4, padx=(40, 0), pady=(0, 0))
+        n_to_ngram_checkbox[4].grid(sticky="W", column=0, row=5, padx=(40, 0), pady=(0, 0))
+        n_to_ngram_checkbox[5].grid(sticky="W", column=0, row=6, padx=(40, 0), pady=(0, 0))
+        n_to_ngram_checkbox[6].grid(sticky="W", column=0, row=7, padx=(40, 0), pady=(0, 0))
```

Comment:
This can be done with a loop:

```python
        for n, checkbox in n_to_ngram_checkbox.items():
            ...
```

Also, don't specify a row, so that it automatically takes the first row with empty space.

Reply:
Thanks, the reason I didn't do this was because of the rows. I did not know they didn't have to be defined. 

However, I'm not sure why I would need the "checkbox" here. I chose to do for n in n_to_ngram_checkbox: n_to_ngram_checkbox[n].grid(...

Seems to work fine.
If I get the checkbox, I wouldn't need "n"

Reply:
You could use n_to_ngram_checkbox.values()